website/source/docs/guides/acl-migrate-tokens.html.md

```diff
@@ -0,0 +1,453 @@
+```
+
+#### Creating Policies
+
+We now have a minimal set of policies to create, with human-readable names. We
+can create each one with something like the following.
+
+```sh
+$ for p in $(ls policies | grep ".hcl"); do \
+  # Extract the hash part of the file name
+  HASH=$(echo "$p" | cut -d - -f 1); \
+  # Extract the name suffix without .hcl
+  NAME=$(echo "$p" | cut -d - -f 2- | cut -d . -f 1); \
+  # Create new policy based on the rules in the file and the name we gave
+  consul acl policy create -name $NAME \
+    -rules "@policies/$p" \
+    -description "Migrated from legacy token"; \
+done
+ID:           da2a9f9b-4e44-13f8-e308-76ce7a8dcb21
+Name:         bar-service
+Description:  Migrated from legacy token
+Datacenters:
+Rules:
+service_prefix "bar" {
+  policy = "write"
+}
+
+ID:           9fbded86-9140-efe4-b661-c8bd07b6c584
+Name:         foo-service
+Description:  Migrated from legacy token
+Datacenters:
+Rules:
+service "foo" { policy = "write" }
+
+```
+
+#### Upgrading Tokens
+
+Finally we can map our existing tokens to those policies using the hash in the
+policy file names. The `-upgrade-legacy` flag removes the token's legacy
+embedded rules at the same time as associating them with the new policies
+created from those rules.
+
+```sh
+$ for id in $LEGACY_IDS; do \
+  NEW_POLICY=$(consul acl translate-rules -token-accessor "$id"); \
+  HASH=$(echo -n "$NEW_POLICY" | shasum | awk '{ print $1 }'); \
+  # Lookup the hash->new policy mapping from the policy file names
+  POLICY_FILE=$(ls policies | grep "^$HASH"); \
+  POLICY_NAME=$(echo "$POLICY_FILE" | cut -d - -f 2- | cut -d . -f 1); \
+  echo "==> Mapping token $id to policy $POLICY_NAME"; \
+  consul acl token update -id $id -policy-name $POLICY_NAME -upgrade-legacy; \
+done
+==> Mapping token 8b65fdf9-303e-0894-9f87-e71b3273600c to policy bar-service
+Token updated successfully.
+AccessorID:   8b65fdf9-303e-0894-9f87-e71b3273600c
+SecretID:     3dbb3981-7654-733a-3475-5ce20fc5a7b9
+Description:
+Local:        false
+Create Time:  0001-01-01 00:00:00 +0000 UTC
+Policies:
+   da2a9f9b-4e44-13f8-e308-76ce7a8dcb21 - bar-service
+==> Mapping token d9deb39b-1b30-e100-b9c5-04aba3f593a1 to policy foo-service
+Token updated successfully.
+AccessorID:   d9deb39b-1b30-e100-b9c5-04aba3f593a1
+SecretID:     5f54733b-4c76-eb74-8781-3550c20f4969
+Description:
+Local:        false
+Create Time:  0001-01-01 00:00:00 +0000 UTC
+Policies:
+   9fbded86-9140-efe4-b661-c8bd07b6c584 - foo-service
+==> Mapping token f2bce42e-cdcc-848d-28ca-cfd0556a22e3 to policy bar-service
+Token updated successfully.
+AccessorID:   f2bce42e-cdcc-848d-28ca-cfd0556a22e3
+SecretID:     f3aaa3e2-2c6f-cf3c-1e86-454de728e8ab
+Description:
+Local:        false
+Create Time:  0001-01-01 00:00:00 +0000 UTC
+Policies:
+   da2a9f9b-4e44-13f8-e308-76ce7a8dcb21 - bar-service
+```
+
+At this point all tokens are upgraded and can use new ACL features while
+retaining the same secret clients are already using.
```

Comment:
Not sure at all on this one, but I think I've had a double take a couple of times when reading this sentence, does it need a `that` in there?

`At this point all tokens are upgraded and can use new ACL features while retaining the same secret that clients are already using.`

Maybe it's just me.

Reply:
It works either way.